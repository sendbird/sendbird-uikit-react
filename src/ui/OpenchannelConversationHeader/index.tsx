import React, { useContext } from 'react';
import './index.scss';

import Avatar from '../Avatar';
import Icon, { IconColors, IconTypes } from '../Icon';
import IconButton from '../IconButton';
import Label, { LabelTypography, LabelColors } from '../Label';
import { LocalizationContext } from '../../lib/LocalizationContext';

interface Props {
  coverImage?: string;
  title?: string;
  subTitle?: string;
  amIOperator?: boolean;
  onActionClick?(): void;
}

export default function OpenchannelConversationHeader({
  coverImage,
  title,
  subTitle,
  amIOperator,
  onActionClick,
}: Props): JSX.Element {
  const { stringSet } = useContext(LocalizationContext);

  return (
    <div className="sendbird-openchannel-conversation-header">
      <div className="sendbird-openchannel-conversation-header__left">
        {
          coverImage ? (
            <Avatar
              className="sendbird-openchannel-conversation-header__left__cover-image"
              src={coverImage}
              alt="channel cover image"
              width="32px"
              height="32px"
            />
          ) : (
            <div
              className="sendbird-openchannel-conversation-header__left__cover-image--icon"
              style={{ width: 32, height: 32 }}
            >
              <Icon
                type={IconTypes.CHANNELS}
                fillColor={IconColors.CONTENT}
                width="18px"
                height="18px"
              />
            </div>
          )
        }
        <Label
          className="sendbird-openchannel-conversation-header__left__title"
          type={LabelTypography.H_2}
          color={LabelColors.ONBACKGROUND_1}
        >
          {title || stringSet.NO_TITLE}
        </Label>
        <Label
          className="sendbird-openchannel-conversation-header__left__sub-title"
          type={LabelTypography.BODY_2}
          color={LabelColors.ONBACKGROUND_2}
        >
          {subTitle || stringSet.NO_TITLE}
        </Label>
      </div>
      <div className="sendbird-openchannel-conversation-header__right">
        <IconButton
          className="sendbird-openchannel-conversation-header__right__trigger"
          width="32px"
          height="32px"
          onClick={onActionClick}
        >
          <Icon
            type={(
              amIOperator
                ? IconTypes.INFO
                : IconTypes.MEMBERS
            )}
            fillColor={IconColors.PRIMARY}
            width="24px"
            height="24px"
          />
        </IconButton>
      </div>
    </div>
  );
}
