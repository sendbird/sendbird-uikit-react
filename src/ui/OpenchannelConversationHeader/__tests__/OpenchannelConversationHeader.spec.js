import React from 'react';
import { render } from '@testing-library/react';

import OpenchannelConversationHeader from "../index";

describe('ui/OpenchannelConversationHeader', () => {
  it('should render default elements', function() {
    const { container } = render(<OpenchannelConversationHeader />);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__cover-image--icon').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__cover-image').length
    ).toBe(0);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__title').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__sub-title').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__right').length
    ).toBe(1);
  });

  it('should render default elements', function() {
    const title = "Open Channel Title";
    const subTitle = "Open Channel Subtitle";
    const { container } = render(
      <OpenchannelConversationHeader
        title={title}
        subTitle={subTitle}
        coverImage="https://static.sendbird.com/sample/user_sdk/user_sdk_20.png"
      />
    );
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__cover-image--icon').length
    ).toBe(0);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__cover-image').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__title').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__left__sub-title').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-openchannel-conversation-header__right').length
    ).toBe(1);
  });

  it('should do a snapshot test of the OpenchannelConversationHeader DOM', function() {
    const { asFragment } = render(
      <OpenchannelConversationHeader
        title="Open Channel Title"
        subTitle="Open Channel Subtitle"
        coverImage="https://static.sendbird.com/sample/user_sdk/user_sdk_20.png"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
