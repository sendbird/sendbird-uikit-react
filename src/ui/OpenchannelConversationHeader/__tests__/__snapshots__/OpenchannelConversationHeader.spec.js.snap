// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/OpenchannelConversationHeader should do a snapshot test of the OpenchannelConversationHeader DOM 1`] = `
<DocumentFragment>
  <div
    class="sendbird-openchannel-conversation-header"
  >
    <div
      class="sendbird-openchannel-conversation-header__left"
    >
      <div
        class="sendbird-openchannel-conversation-header__left__cover-image sendbird-avatar"
        role="button"
        style="height: 32px; width: 32px; z-index: 0; left: 0px;"
        tabindex="0"
      >
        <div
          class="sendbird-avatar-img sendbird-image-renderer"
          style="width: 100%; min-width: min(400px, 32px); max-width: 400px;"
        >
          <div
            class="sendbird-image-renderer__image"
            style="width: 100%; min-width: min(400px, 32px); max-width: 400px; position: absolute; background-repeat: no-repeat; background-position: center; background-size: cover; background-image: url(https://static.sendbird.com/sample/user_sdk/user_sdk_20.png);"
          />
          <img
            alt="channel cover image"
            class="sendbird-image-renderer__hidden-image-loader"
            src="https://static.sendbird.com/sample/user_sdk/user_sdk_20.png"
          />
        </div>
      </div>
      <span
        class="sendbird-openchannel-conversation-header__left__title sendbird-label sendbird-label--h-2 sendbird-label--color-onbackground-1"
      >
        Open Channel Title
      </span>
      <span
        class="sendbird-openchannel-conversation-header__left__sub-title sendbird-label sendbird-label--body-2 sendbird-label--color-onbackground-2"
      >
        Open Channel Subtitle
      </span>
    </div>
    <div
      class="sendbird-openchannel-conversation-header__right"
    >
      <button
        class="sendbird-openchannel-conversation-header__right__trigger sendbird-iconbutton "
        style="height: 32px; width: 32px;"
        type="button"
      >
        <span
          class="sendbird-iconbutton__inner"
        >
          <div
            class=" sendbird-icon sendbird-icon-members sendbird-icon-color--primary"
            role="button"
            style="width: 24px; min-width: 24px; height: 24px; min-height: 24px;"
            tabindex="0"
          >
            <test-file-stub />
          </div>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
